pkg/test/framework/components/ingress/ingress.go

```diff
@@ -21,6 +21,15 @@ import (
 	"istio.io/istio/pkg/test/framework/resource"
 )
 
+// IgType defines ingress gateway type
+type IgType int
+
+const (
+	PlainText IgType = 0
+	TLS       IgType = 1
+	Mtls      IgType = 2
```

Comment:
Are there any off-shelf proto enum definition from Istio API that we can use?

Reply:
I don't find a proto enum from Istio API that could be used for ingress gateway type.